### ISSUE TITLE
When trying to do the examples: Parameter should have an immutability of at least "Immutable" (actual: "Unknown")

### ISSUE THREAD

**Issue by atmosbear** [eslint-functional/eslint-plugin-functional#676]
## Bug Report

I'm getting 'Parameter should have an immutability of at least "Immutable" (actual: "Unknown")." for every function within the [example page]( https://github.com/eslint-functional/eslint-plugin-functional/blob/v5.0.8/docs/rules/prefer-immutable-types.md) where it says it's supposed to be "correct" with a :white_check_mark: 

I'm running Linux on VSCode 1.79.2

### Expected behavior

It should not display an error :)

### Actual behavior

it displays errors :( (lol) see image below

### Steps to reproduce
output:
![image](https://github.com/eslint-functional/eslint-plugin-functional/assets/111397932/14232542-efda-45ce-8f3f-e8e80efdb1d1)

package.json:
```json
{
	"dependencies": {
		"@typescript-eslint/eslint-plugin": "^5.60.0",
		"vite": "^4.3.9"
	},
	"devDependencies": {
		"eslint": "^8.43.0",
		"eslint-plugin-functional": "^5.0.8"
	}
}

```
.eslintrc:
```js
{
 "parser": "@typescript-eslint/parser",
 "extends": [],
 "plugins": [
  "functional",
  "@typescript-eslint"
 ],
 "rules": {
  "functional/prefer-immutable-types": "error"
 }
}
```
index.ts:
```ts
// the same as the example at  https://github.com/eslint-functional/eslint-plugin-functional/blob/v5.0.8/docs/rules/prefer-immutable-types.md:

function array1(arg: ReadonlyArray<string>) {}
function array2(arg: ReadonlyArray<ReadonlyArray<string>>) {}
function array3(arg: readonly [string, number]) {}
function array4(arg: readonly [ReadonlyArray<string>, number]) {}
// the above examples work the same if you use ReadonlyArray<T> instead

function object1(arg: { readonly prop: string }) {}
function object2(arg: { readonly prop: string; readonly prop2: string }) {}
function object3(arg: { readonly prop: { readonly prop2: string } }) {}
// the above examples work the same if you use Readonly<T> instead

interface CustomArrayType extends ReadonlyArray<string> {
  readonly prop: string;
}
function custom1(arg: Readonly<CustomArrayType>) {}
// interfaces that extend the array types are not considered arrays, and thus must be made readonly.

interface CustomFunction {
  (): void;
  readonly prop: string;
}
function custom2(arg: CustomFunction) {}

function union(arg: ReadonlyArray<string> | ReadonlyArray<number[]>) {}

function primitive1(arg: string) {}
function primitive2(arg: number) {}
function primitive3(arg: boolean) {}
function primitive4(arg: unknown) {}
function primitive5(arg: null) {}
function primitive6(arg: undefined) {}
function primitive7(arg: any) {}
function primitive8(arg: never) {}
function primitive9(arg: string | number | undefined) {}

function fnSig(arg: () => void) {}

enum Foo { a, b }
function enum1(arg: Foo) {}

function symb1(arg: symbol) {}
const customSymbol = Symbol('a');
function symb2(arg: typeof customSymbol) {}

// function types
interface Foo1 {
  (arg: ReadonlyArray<string>): void;
}
interface Foo2 {
  new (arg: ReadonlyArray<string>): void;
}
const x = { foo(arg: ReadonlyArray<string>): void; };
function foo(arg: ReadonlyArray<string>);
type Foo3 = (arg: ReadonlyArray<string>) => void;
interface Foo4 {
  foo(arg: ReadonlyArray<string>): void;
}

```

Just for clarity, it does this whether or not:
1. I use ReadonlyArray<whatever> instead of readonly or vice versa. 
2. all other extensions (not ESLint) are disabled
3. I move the `/* eslint functional/prefer-immutable-types: "error" */` line into the index.ts file and remove the rule in .eslintrc
4. I have other rules, plugins, etc

It's like it doesn't recognize it as immutable.

**Comment by pegak** [comment]
I have the same problem with other things. Didn't you (by any chance) upgraded to v6?

This is my error message:
`3:31  Error: Parameter should have an immutability of at least "ReadonlyShallow" (actual: "Unknown").  functional/prefer-immutable-types`

**Comment by atmosbear** [comment]
> I have the same problem with other things. Didn't you (by any chance) upgraded to v6?
> 
> This is my error message: `3:31 Error: Parameter should have an immutability of at least "ReadonlyShallow" (actual: "Unknown"). functional/prefer-immutable-types`

I'm so sorry, I don't use this plugin anymore and I completely forgot whether or not I had upgraded. Sorry!!

**Comment by RebeccaStevens** [comment]
You haven't enabled type-checking in your `.eslintrc` file.

Add this to your config file:

```jsonc
"parserOptions": {
  "project": "./tsconfig.json"
}
```

See https://github.com/eslint-functional/eslint-plugin-functional/blob/main/GETTING_STARTED.md#with-typescript
